spec.bs

```diff
@@ -173,18 +173,20 @@ are:
       1. [=list/Append=] |igAd| to |interestGroup|'s |interestGroupField|.
 1. If |interestGroup|'s [=interest group/estimated size=] is greater than 50 KB, then
   [=exception/throw=] a {{TypeError}}.
-1. Let |permission| be the result of [=checking interest group permissions=] with 
-  |ownerUrl|'s [=url/origin=], |frameOrigin|, and true.
-1. If |permission| is false, then [=exception/throw=] a "{{NotAllowedError}}" {{DOMException}}.
 1. Let |p| be [=a new promise=].
 1. Let |queue| be the result of [=starting a new parallel queue=].
 1. [=parallel queue/enqueue steps|Enqueue the following steps=] to |queue|:
-  1. [=Queue a task=] to [=resolve=] |p| with `undefined`.
-  1. If the browser is currently storing an interest group with `owner` and `name` that matches
-    |interestGroup|, then remove the currently stored one.
-  1. Set |interestGroup|'s [=interest group/joining origin=] to [=this=]'s
-    [=relevant settings object=]'s [=environment/top-level origin=]
-  1. Store |interestGroup| in the browser’s [=interest group set=].
+  1. Let |permission| be the result of [=checking interest group permissions=] with 
+    |ownerUrl|'s [=url/origin=], |frameOrigin|, and true.
+  1. If |permission| is false, then [=queue a task=] to [=reject=] |p| with a
+     "{{NotAllowedError}}" {{DOMException}} and do not run the remaining steps.

```

Comment:
btw, you can just say "return a rejected promise with NotAllowedError" (https://webidl.spec.whatwg.org/#a-promise-rejected-with), and you don't have to say "do not run the remaining steps".

Reply:
I'm confused, though -- the `Return |p|.` occurs later, on the main thread. You can't return a promise in a parallel queue, right?

Reply:
I would like to defer to @domfarolino - I'm not totally sure.

Reply:
Maybe we just say "resolve" instead of "return".

Reply:
I think what you have is good enough here.